packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -116,18 +116,20 @@ export class CallContext {
   // make sure the state is clean because any left over state (if previous CallAgentDeclarative was disposed) may be
   // invalid.
   public clearCallRelatedState(): void {
-    this.modifyState((draft: CallClientState) => {
-      draft.calls = {};
-      draft.incomingCalls = {};
-      draft.callsEnded = {};
-      draft.incomingCallsEnded = {};
-    });
+    this.modifyState((draft: CallClientState) => ({
+      ...draft,
+      calls: {},
```

Comment:
Hmmm, this will break the functionality of immutable and make it not working as intended, draft.[property] = [value] is not just a simple assignment expression, it is a setter function overriden by immer library, not calling it will cause immer.js cannot detect some deep changes in the state tree, which might cause some unexpected error/perf regressions

Reply:
I was reading the docs on immer, it seemed like this reassignment would also work. if the state of the draft is updated in any way, when the function terminates it updates the newState to what was changed. or am I misunderstanding something?